core/src/main/scala/org/bitcoins/core/package.scala

```diff
@@ -31,6 +32,19 @@ package object core {
     }
   }
 
+  implicit class bigIntegerUtil(private val bigInt: BigInteger) extends AnyVal {
+
+    implicit def intExact: Int = {
+      if (bigInt.bitLength() <= 31) bigInt.intValue()
```

Comment:
could we just use a `require(bigInt == bigInt.intValue())` 

Same for below

Reply:
These are just copied from the actual java implementations, I am guessing this is more effienct